apps/email/js/cards/message_list.js

```diff
@@ -751,6 +751,21 @@ return [
       cards.removeCardAndSuccessors(this, 'animate');
     },
 
+    onClearSearch: function() {
+      this.curPhrase = '';
+      this.curFilter = 'all';
+
+      this.curFolder = model.folder;
+      this.vScroll.clearDisplay();
+
+      headerCursor.startSearch(this.curPhrase, {
+        author: this.curFilter,
+        recipients: this.curFilter,
+        subject: this.curFilter,
+        body: this.curFilter
+      });
+    },
```

Comment:
I think it is just enough to call showSearch with an empty string, that way we can avoid some duplication. So, something just like `this.showSearch('', this.curFilter);` looks like all that is needed.